core/shared/src/main/scala/zio/Fiber.scala

```diff
@@ -824,11 +824,11 @@ object Fiber extends FiberPlatformSpecific {
   def unsafeCurrentFiber(): Option[Fiber[Any, Any]] =
     Option(_currentFiber.get)
 
-  private[zio] def newFiberId(): FiberId =
-    FiberId(java.lang.System.currentTimeMillis(), _fiberCounter.getAndIncrement())
+  private[zio] def newFiberId(): FiberId.Runtime =
+    FiberId.Runtime((java.lang.System.currentTimeMillis / 1000).toInt, _fiberCounter.getAndIncrement())
```

Comment:
Don't we need more precision here?

Reply:
BTW wouldn't it make more sense to use jvm time instead of wall-clock time here?

Reply:
Interesting post [here](http://pzemtsov.github.io/2017/07/23/the-slow-currenttimemillis.html) on the performance of `currentTimeMillis`. 

We'd like to use the fastest method of measuring some notion of time, rather than the most accurate one. 

This data will only be used by metrics (as well as to make an `AtomicInteger`'s wraparound issues irrelevant).

Reply:
From what I gather:

1. `nanoTime` is extremely slow (expected)
2. `currentTimeMillis` is fast on Windows, and slow (but not extremely slow) on Linux

An OpenJDK issue was closed as "Won't fix" regarding (2).

I think we can use `currentTimeMillis` here and fix it later.

"second time", which is achieved by `/ 1000` is more than sufficient because in order for that to lead to uniqueness issues via `AtomicInteger` sequence number, we would have to be generating 2,147,483,647 fibers per second, which of course can't happen.

Reply:
Interesting discussion. Thought neither of you realised, that `FiberId.Runtime` changed the order of id and timestamp and this impl. is therefore a bug.
I already fixed it in [this pr](https://github.com/zio/zio/pull/5916) so don't worry 😃